petab/v2/models/_sbml_utils.py

```diff
@@ -0,0 +1,52 @@
+"""Private utility functions for SBML handling."""
+
+import libsbml
+
+retval_to_str = {
+    getattr(libsbml, attr): attr
+    for attr in (
+        "LIBSBML_DUPLICATE_OBJECT_ID",
+        "LIBSBML_INDEX_EXCEEDS_SIZE",
+        "LIBSBML_INVALID_ATTRIBUTE_VALUE",
+        "LIBSBML_INVALID_OBJECT",
+        "LIBSBML_INVALID_XML_OPERATION",
+        "LIBSBML_LEVEL_MISMATCH",
+        "LIBSBML_NAMESPACES_MISMATCH",
+        "LIBSBML_OPERATION_FAILED",
+        "LIBSBML_UNEXPECTED_ATTRIBUTE",
+        "LIBSBML_PKG_UNKNOWN",
+        "LIBSBML_PKG_VERSION_MISMATCH",
+        "LIBSBML_PKG_CONFLICTED_VERSION",
+    )
+}
+
+
+def check(res: int):
+    """Check the return value of a libsbml function that returns a status code.
+
+    :param res: The return value to check.
+    :raises RuntimeError: If the return value indicates an error.
+    """
+    if res != libsbml.LIBSBML_OPERATION_SUCCESS:
+        raise RuntimeError(f"libsbml error: {retval_to_str.get(res, res)}")
+
+
+def add_sbml_parameter(
+    model: libsbml.Model,
+    id_: str = None,
```

Comment:
IDs are required for SBML parameters
```suggestion
    id_: str,
```

Reply:
To have a valid model, yes. But during construction it can be unset.

Reply:
As far as I can tell, we have no use case for leaving it unset? All uses in this PR appear to set the ID.

Reply:
Potentially for testing invalid models, but not right now. It was meant as a generic wrapper for the most commonly used setXXX. Either way works for me :man_shrugging: